dockerfiles/base/scripts/base/commands/cmd_version.sh

```diff
@@ -20,16 +20,66 @@ cmd_version() {
   else
     text "Your installed version is '<not-installed>'.\n"
   fi
-  text "Available on DockerHub:\n"
-  VERSION_LIST_JSON=$(curl -s https://hub.docker.com/v2/repositories/${CHE_IMAGE_NAME}/tags/)
-  NUMBER_OF_VERSIONS=$(echo $VERSION_LIST_JSON | jq '.count')
+
+  text "\n"
+
+  if is_offline; then
+    text "Available on DockerHub: offline mode\n"
+  else  
+    text "Available on DockerHub:\n"
+
+    local VERSION_LIST_JSON=$(curl -s https://hub.docker.com/v2/repositories/${CHE_IMAGE_NAME}/tags/)
+    local NUMBER_OF_VERSIONS=$(echo $VERSION_LIST_JSON | jq '.count')
+
+    DISPLAY_LIMIT=10
+    if [ $DISPLAY_LIMIT -gt $NUMBER_OF_VERSIONS ]; then 
+      DISPLAY_LIMIT=$NUMBER_OF_VERSIONS
+    fi
+
+    COUNTER=0
+    while [ $COUNTER -lt $DISPLAY_LIMIT ]; do
+      TAG=$(echo $VERSION_LIST_JSON | jq ".results[$COUNTER].name")
+      text "  ${TAG//\"}\n"
+      let COUNTER=COUNTER+1 
+    done
+
+    if [ $NUMBER_OF_VERSIONS -gt $DISPLAY_LIMIT ]; then
+      OLDER_VERSION=$(echo $VERSION_LIST_JSON | jq '.next')
+      text "  See older versions at: $OLDER_VERSION\n"
+    fi
+  fi
+}
+
+# Compares $1 version to the first 10 versions listed as tags on Docker Hub
+# Returns "" if $1 is newest, otherwise returns the newest version available
+compare_versions() {
+
+  local VERSION_LIST_JSON=$(curl -s https://hub.docker.com/v2/repositories/${CHE_IMAGE_NAME}/tags/)
+  local NUMBER_OF_VERSIONS=$(echo $VERSION_LIST_JSON | jq '.count')
+
+  DISPLAY_LIMIT=10
+  if [ $DISPLAY_LIMIT -gt $NUMBER_OF_VERSIONS ]; then 
+    DISPLAY_LIMIT=$NUMBER_OF_VERSIONS
+  fi
+
+  # Strips off -M#, -latest version information
+  BASE_VERSION=$(echo $1 | cut -f1 -d"-")
```

Comment:
The milestone and rc markers were causing failures on version comparison tests.   And so for this exercise I figured that the version comparison test did not need to account for the milestone markers until we consider them again in the future.

So this strips off the milestone and just uses the numerical comparator.